client/my-sites/controller.js

```diff
@@ -537,6 +537,14 @@ export function siteSelection( context, next ) {
 		'1' === context.query?.unlinked && context.pathname.match( /^\/checkout\/[^/]+\/jetpack_/i );
 
 	const shouldRenderNoSites = ! context.section.enableNoSites && ! isUnlinkedCheckout;
+	const isFromMyJetpackLoggedOut = context.query?.source === 'my-jetpack' && ! currentUser;
+
+	// Immediately continue if the user is logged out and coming from My Jetpack
+	// so it gets picked up by the checkout middleware and redirects to logged-out checkout
+	if ( isFromMyJetpackLoggedOut ) {
```

Comment:
NIT The same, looks redundant to have separate variable and the comment. I would not create at least separate variable, but it's not critical, up to you.